### PR TITLE
Fix fetching search results and handle fetching errors

### LINKED ISSUE #2588: Improve error handling in the media store
## Problem

We get a lot of errors related to media fetching, but the store hides all the error details so we cannot examine and fix them.

## Description

We should refactor the error handling in the media store. The errors should be sent to Sentry with all of the details.
We should also show an error page when there are errors in fetching search results to prevent the users from sending other requests by clicking on the search results.

## Additional context

The Sentry errors show that the users get rate-limited and then navigate to the single results, which throws an error:
https://openverse.sentry.io/share/issue/188ef4f5764a4eb0b6c5a81abc71e4df/

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #2588 by @obulat

## Description

This PR moves the fetching of search results from the `asyncData` to the `search` middleware and the `useFetch` hook in the `search.vue` page. 

### Fetching search results on the server
Just like in the single result pages, on the server, we make sure that we can fetch the data or redirect to the Nuxt error page. In the `search.vue` page, we set `fetchOnServer` to false so that the data that was fetched on the server (if the search page was opened directly, when you go to https://openverse.org/search/?q=cat) isn't refetched when the page is hydrating.

### Fetching search results on the client
If we navigate to the search page from other pages (homepage or 404 page), the `useFetch`calls `fetchMedia` in `search.vue` to get the search results. If the route changes (either the path, i.e. when the search type changes, or the query, i.e., when the search term, or filters change), the router watch calls `fetchMedia` in `search.vue`.

### Updating the provider filters
Previously, `asyncData` updated the search state and provider filters. Here, this is moved to the `search` middleware.

### Fetching errors
When there are no search results for a query, we receive a 200 response with an empty result list. In the `media` store, we set ~~`fetchingError` to 404 status in case there are no results, to show the 404 page.~~ `fetchingError.message` to `NO_RESULT` to show the No results page.

#### All content errors
This PR updates the handling of the "All content" errors: if there is even a single 429 error, we set the combined `allMediaError` status to 429, because it means that all client searches will be throttled.
~~`allMediaError` is set to 404 status only if the `fetchingError`s for all media types is 404, so we only show a 404 page for "All content" if both audio and image had no results.~~  After @dhruvkb's request for change, I've updated `allMediaError` to set a `NO_RESULT` error message if there are no results for all supported media types. Otherwise, we show the available media type results.

## Testing Instructions

Searching from the homepage and 404 page, as well as from the search page's header should work correctly.
Clicking "Load more" should load the next page of the results for all search types.

For all media, check cases with only one type of results, such as "ecommerce image". 
Also, try queries that don't have any results (such as "querywithnoresults").

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by AetherUnbound — APPROVED** [review]
These changes generally make sense to me! I followed the testing instructions and everything worked as expected.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
LGTM, works well!

> In the `media` store, we set `fetchingError` to 404 status in case there are no results, to show the 404 page.

This seems wrong to me. If we have no results for a query, that does not imply 404 because the `/search` page itself still exists.

File: frontend/src/pages/search.vue [hunk 1]
```diff
@@ -79,39 +86,70 @@ export default defineComponent({
     const { resultCount, fetchState, resultItems } = storeToRefs(mediaStore)
 
     const needsFetching = computed(() =>
-      Boolean(
-        supported.value && !resultCount.value && searchTerm.value.trim() !== ""
-      )
+      Boolean(supported.value && !resultCount.value && searchTerm.value !== "")
```
[the remark this answers is not in the document]
Comment (obulat) [reply frontend/src/pages/search.vue:89]: We trim the `searchTerm` value when we set it in the store, so we don't need to do it here.

**Comment by obulat** [comment]
> This seems wrong to me. If we have no results for a query, that does not imply 404 because the /search page itself still exists.
Right, I haven't thought about it. Thank you, Dhruv!

**Comment by obulat** [comment]
> LGTM, works well!
> 
> > In the `media` store, we set `fetchingError` to 404 status in case there are no results, to show the 404 page.
> 
> This seems wrong to me. If we have no results for a query, that does not imply 404 because the `/search` page itself still exists.

I updated the PR to set the error to `{message: NO_RESULTS}` when we have no results so that we show the client-side No results page instead of the Nuxt yellow error page.

**Review by dhruvkb — APPROVED** [review]
LGTM, thanks! I appreciate your quick turnaround on the requested changes.

**Pull request merged**